### PR TITLE
feat: add query call builder, call_raw on sync calls and wallet canister interface

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: ic-utils/src/call.rs [hunk 1]
```diff
@@ -132,37 +147,42 @@ where
...
-    where
-        R: for<'de> ArgumentDecoder<'de> + Send + Sync,
-    {
-        self.agent
-            .query(&self.canister_id, &self.method_name)
-            .with_arg(&self.method_name)
-            .call()
-            .await
-            .and_then(|r| decode_args(&r).map_err(|e| AgentError::CandidError(Box::new(e))))
+    async unsafe fn call_raw(self) -> Result<Vec<u8>, AgentError> {
```
Comment (ericswanson-dfinity) [inline ic-utils/src/call.rs:167]: Is the whole reason for `unsafe` here to discourage direct calls to call_raw?

It seems like as a side-effect, this function could accumulate unsafe code.

Could we change this function to a one-liner that forwards to a function that is not marked `unsafe` ?

Comment (ericswanson-dfinity) [reply ic-utils/src/call.rs:167]: Or is there some other way to hide `call_raw` ?

File: ic-utils/src/interfaces/wallet.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    ) -> impl 'agent + AsyncCall<()> {
+        self.update_("authorize").with_arg(custodian).build()
+    }
+
+    /// Deauthorize a custodian.
+    pub fn deauthorize<'canister: 'agent>(
+        &'canister self,
+        custodian: Principal,
+    ) -> impl 'agent + AsyncCall<()> {
+        self.update_("authorize").with_arg(custodian).build()
```
Comment (ericswanson-dfinity) [inline ic-utils/src/interfaces/wallet.rs:144]: should this be the following?
```suggestion
        self.update_("deauthorize").with_arg(custodian).build()
```

File: ic-utils/src/call.rs (hunk 1 above)
Comment (hansl) [reply ic-utils/src/call.rs:167]: Feature flag would be another way. You have to "opt in" at the crate level, which is fine I guess.

Comment (hansl) [reply ic-utils/src/call.rs]: Done. PTAL

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**